### DISCUSSION TITLE
How to connect to MS Sql Server?

### DISCUSSION THREAD

**Discussion by SwissMaWi** [sqlpage/SQLPage#90] [Q&A] [answered]
I searched for a while but cannot seem to find instructions on how to set the database_url in the sqlpage.json to connect to Microsoft SQL Server.

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !

You have to specify a database connection string of the form `mssql://username:password@localhost/YourDatabaseName` in sqlpage.json to connect to your databse. If the database is not running on the local host, replace `localhost` with the ip or domain name of the database server. If you are using a named instance, you will additionally have to [specify a port](https://github.com/lovasoa/SQLpage/issues/92).

**Reply by SwissMaWi** [reply]
Thanks. That will help. Would be cool if this information was in the documentation.
And thanks a lot for all the work.
I am trying to build a dashboard using SQLpage.
One of the databases is mySql only reachable via IP over SSH... can I also connect to that?

**Reply by lovasoa** [reply, marked as answer]
Yes, you can connect to it, but you'll need to open the ssh connection outside of sqlpage. If your mysql server is accessible by ssh on `your-server` and uses the default port `3306`:

```bash
ssh -L 3306:localhost:3306 your-server
```

Then in `sqlpage.json` : 

```json
{
    "database_url": "mysql://user:password@localhost/your_database"
}
```
